spring/data-core-jakarta/src/main/kotlin/com/linecorp/kotlinjdsl/spring/data/SpringDataQueryFactoryExtensions.kt

```diff
@@ -9,6 +9,10 @@ inline fun <reified T> SpringDataQueryFactory.singleQuery(
     noinline dsl: SpringDataCriteriaQueryDsl<T>.() -> Unit
 ): T = selectQuery(T::class.java, dsl).singleResult
 
+inline fun <reified T> SpringDataQueryFactory.singleOrNullQuery(
+    noinline dsl: SpringDataCriteriaQueryDsl<T>.() -> Unit
+): T? = selectQuery(T::class.java, dsl).resultList.firstOrNull()
```

Comment:
When there are more than one element in the list result, I think that throwing an exception can reduce the confusion on the interface.

This method has the name `single`. In the current interface, the name `single` has the same meaning as JPA's `single`, and the meaning contained in `single` means that if there are more than one element, an exception will be occurred.

It would be nice if we threw the NonUniqueResultException, but this exception exists in each of javax and jakarta, so it's hard to use here.

Therefore, when there are more than one element in the list result, I think throwing with a predefined exception such as 'IllegalStateException()' is the best thing in the current module structure.

-- korean
List에 아이템이 여러개 있을 경우 예외를 발생시키는 것이 사용자에게 혼란을 줄일 수 있어 더 좋은 인터페이스라고 생각합니다.

메소드에 single 이라는 이름이 포함되어 있습니다. 현재 인터페이스에서 single이라는 이름은 JPA의 single을 의미하며, 이것은 만약 여러개의 list 아이템이 있을 경우 예외를 발생시키겠다는 것을 의미합니다.

NonUniqueResultException을 사용하면 좋겠지만 이 예외는 javax와 jakarta 각각에 존재하기 때문에 사용하기 어렵습니다.

그렇기 때문에 리스트에 아이템이 여러개 있을 경우 `IllegalStateException()`과 같이 미리 정의된 예외로 던지는 정도가 현재 모듈 구조에서의 최선이라고 생각합니다.